contracts/interfaces/IEscrow.sol

```diff
@@ -8,6 +8,13 @@ interface IEscrow {
         address[] rewardTokens;
     }
 
+    struct Deposit {
+        uint amount;
+        bool isERC4626;
+        address asset; // eip4626 asset else the erc20 token itself
```

Comment:
No, `tokensUsed` tracks the actual token deposited and then we track the underlying. This is useful because someone could deposit USDC and a share token that is backed by USDC, we want to be able to go through both types of tokens.